lib/queue.js

```diff
@@ -440,7 +426,7 @@ Queue.prototype._finishJob = function (err, data, job, cb) {
   };
 
   if (err) {
-    job.options.stacktraces.unshift(err.stack);
+    if (err.stack) job.options.stacktraces.unshift(err.stack);
```

Comment:
This should `unshift` something onto the stacktraces array so we have a measure of the number of attempts made, and some record that an error was thrown that didn't have a stack.

`unshift(err ? err.stack || err.message || err : null)`